### ISSUE TITLE
Training the model on custom dataset

### ISSUE THREAD

**Issue by harshithdwivedi** [javier2491/FaceQnet#14]
Is there a way for me to train the model on my custom dataset?
For instance, I have faces labeled as "focused", "almost_focused" and "out_of_focus" and I have assigned a score of 0,0.5 and 1 to each of them respectively.

Can you share the scripts that can be used to train the model? Because I noticed that the pre-trained model doesn't tend to give accurate results.

For instance, I noticed that the model gave a score of 0.5-0.55 to a lot of out-of-focus faces.

**Comment by amyburden** [comment]
I also find similar problem

**Comment by javier-hernandezo** [comment]
Hello, 

You can train the model using the scores you gave to the different types of focus (0, 0.5 and 1). There should not be any problem. 
Just use the pretrained FaceQnet model as your initial model, and retrain it (entirely or just the last layers) with your groundtruth database.

If it is not giving you accurate results, it may be that you are having problems with the training parameters (e.g., learning rate, dropout rate, etc), your groundtruth is not really well labeled, or you do not have enough training data.
If you give me more details about all these points maybe I could help you more.

Regards

**Comment by amyburden** [comment]
> Hello,
> 
> You can train the model using the scores you gave to the different types of focus (0, 0.5 and 1). There should not be any problem.
> Just use the pretrained FaceQnet model as your initial model, and retrain it (entirely or just the last layers) with your groundtruth database.
> 
> If it is not giving you accurate results, it may be that you are having problems with the training parameters (e.g., learning rate, dropout rate, etc), your groundtruth is not really well labeled, or you do not have enough training data.
> If you give me more details about all these points maybe I could help you more.
> 
> Regards

I believe that it is caused by inconsistency of ground truth. When I use faceqnet output as ground truth, this will not appear. However, I use normalized distance as ground truth. Model cannot learn anything from it.
